js/bitforex.js

```diff
@@ -0,0 +1,605 @@
+        }
+        return result;
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        await this.loadMarkets ();
+        let market = this.markets[symbol];
+        let request = {
+            'symbol': market.id,
+        };
+        let response = await this.publicGetApiV1MarketTicker (this.extend (request, params));
+        let data = response['data'];
+        let timestamp = this.safeInteger (data, 'date');
+        return {
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': this.safeFloat (data, 'high'),
+            'low': this.safeFloat (data, 'low'),
+            'bid': this.safeFloat (data, 'buy'),
+            'bidVolume': undefined,
+            'ask': this.safeFloat (data, 'sell'),
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': undefined,
+            'close': this.safeFloat (data, 'last'),
+            'last': this.safeFloat (data, 'last'),
+            'previousClose': undefined,
+            'change': undefined,
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': this.safeFloat (data, 'vol'),
+            'quoteVolume': undefined,
+            'info': response,
+        };
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let marketId = this.marketId (symbol);
+        let request = {
+            'symbol': marketId,
+        };
+        if (typeof limit !== 'undefined') {
+            request['size'] = limit;
+        }
+        let response = await this.publicGetApiV1MarketDepth (this.extend (request, params));
+        let data = response['data'];
+        let timestamp = response['time'];
+        let bidsKey = 'bids';
+        let asksKey = 'asks';
+        let priceKey = 'price';
+        let amountKey = 'amount';
+        let orderbook = this.parseOrderBook (data, timestamp, bidsKey, asksKey, priceKey, amountKey);
+        return orderbook;
+    }
+
+    async fetchMyTrades (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        let requestClosed = {
+            'symbol': this.marketId (symbol),
+            'state': 1,
+        };
+        let responseClosed = await this.privatePostApiV1TradeOrderInfos (this.extend (requestClosed, params));
+        let closedData = [];
+        if ('data' in responseClosed) {
+            closedData = responseClosed['data'];
+        }
+        let completedOrders = [];
+        for (let i = 0; i < closedData.length; i++) {
+            let current = closedData[i];
+            if (current['orderState'] === 2) {
+                completedOrders.push (current);
+            }
+        }
+        return this.parseOrders (completedOrders, market, since, limit);
```

Comment:
The API doesn't seem to allow us to query for a user's completed trades. Instead I've made a request for all closed orders and filtered for completed orders. (`orderState === 2`).

While this works, it's not currently in a proper format. I'm returning `this.parseOrders()` instead of `this.parseTrades()`. If this approach is satisfactory to you I'll rework `parseTrade()` to also parse orders correctly into the the trade format:

```
{
            'info': 
            'id': 
            'timestamp': 
            'datetime': 
            'symbol': ,
            'type': ,
            'side': ,
            'price': ,
            'amount': ,
            'cost': ,
            'order': ,
            'fee': ,
};
```

Reply:
I would rather not include this type of emulated implementation (as explained in my previous comment). Hope you don't mind keeping this particular one within userland for now?

Reply:
Not at all, I can remove this method and I'll mark `'fetchMyTrades': false` in the description.

Reply:
@JoshVarty that would be awesome!) thx so much for your understanding and for your involvement! )